pkg/plugin/processor/builtin/avro/decode.go

```diff
@@ -0,0 +1,206 @@
+	return sdk.Specification{
+		Name:    "avro.encode",
+		Summary: "Decodes a field's raw data in the Avro format",
+		Description: `The processor takes raw data (bytes or a Base64-encoded string) in the specified field and decodes 
+it from the [Avro format](https://avro.apache.org/) into structured data. It extracts the schema ID from the data, 
+downloads the associated schema from the [schema registry](https://docs.confluent.io/platform/current/schema-registry/index.html)
+and decodes the payload. The schema is cached locally after it's first downloaded. 
+
+Currently, the processor only supports the Avro format. If the processor encounters structured data or the data 
+can't be decoded it returns an error.
+
+This processor is the counterpart to 'avro.encode'.`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: decodeConfig{}.Parameters(),
+	}, nil
+}
+
+func (p *decodeProcessor) Configure(ctx context.Context, m map[string]string) error {
+	cfg, err := parseDecodeConfig(ctx, m)
+	if err != nil {
+		return cerrors.Errorf("invalid config: %w", err)
+	}
+
+	p.cfg = cfg
+
+	return nil
+}
+
+func (p *decodeProcessor) Open(ctx context.Context) error {
+	client, err := schemaregistry.NewClient(p.logger, p.cfg.ClientOptions()...)
+	if err != nil {
+		return cerrors.Errorf("could not create schema registry client: %w", err)
+	}
+	p.decoder = schemaregistry.NewDecoder(client, p.logger, &sr.Serde{})
+
+	return nil
+}
+
+func (p *decodeProcessor) Process(ctx context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		proc, err := p.processRecord(ctx, rec)
+		if err != nil {
+			return append(out, sdk.ErrorRecord{Error: err})
+		}
+
+		out = append(out, proc)
+	}
+
+	return out
+}
+
+func (p *decodeProcessor) processRecord(ctx context.Context, rec opencdc.Record) (sdk.ProcessedRecord, error) {
+	field, err := p.cfg.fieldResolver.Resolve(&rec)
+	if err != nil {
+		return nil, cerrors.Errorf("failed resolving field: %w", err)
+	}
+
+	data, err := p.rawData(field.Get())
+	if err != nil {
+		return nil, cerrors.Errorf("failed getting raw data: %w", err)
+	}
+
+	rd, err := p.decoder.Decode(ctx, data)
+	if err != nil {
+		return nil, cerrors.Errorf("failed encoding data: %w", err)
+	}
+
+	err = field.Set(rd)
+	if err != nil {
+		return nil, cerrors.Errorf("failed setting the decoded value: %w", err)
+	}
+	return sdk.SingleRecord(rec), nil
+}
+
+func (p *decodeProcessor) rawData(data any) (opencdc.RawData, error) {
+	switch v := data.(type) {
+	case opencdc.RawData:
+		return v, nil
+	case []byte:
+		return v, nil
+	case string:
+		decoded := make([]byte, base64.StdEncoding.DecodedLen(len(v)))
+		n, err := base64.StdEncoding.Decode(decoded, []byte(v))
+		if err != nil {
+			return nil, cerrors.Errorf("couldn't base64-decode data: %w", err)
+		}
+		return opencdc.RawData(decoded[:n]), nil
```

Comment:
Hmm, so the assumption is that the string is coming from a JSON encoded record? 🤔 Is there a use case when this string wouldn't be base64 encoded?

Reply:
We decided to put this functionality in a separate processor [base64 encode/decode](https://github.com/ConduitIO/conduit/pull/1412).

Reply:
For the record, Lovro and I chatted about this. While it's likely that a string that represents Avro bytes is a Base64 encoding, it doesn't have to be. However, that's not always the case, meaning the processor will not always be usable out of the box. We will provide a Base64 encoder and decoder that can be used in these case.